src/Aspire.Hosting.Azure/Provisioning/Internal/CredentialProviderDetector.cs

```diff
@@ -0,0 +1,253 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Azure.Provisioning.Internal;
+
+/// <summary>
+/// Detects available Azure credential providers in the user's environment.
+/// </summary>
+internal sealed class CredentialProviderDetector(ILogger<CredentialProviderDetector> logger)
```

Comment:
Simplify this logic and just use the TokenCredential and don't check for stuff on the PATH or by running commands. Just delegate.